src/fmeval/transforms/transform_pipeline.py

```diff
@@ -0,0 +1,67 @@
+import ray.data
+from typing import List, Union
+
+from fmeval.constants import TRANSFORM_PIPELINE_MAX_SIZE
+from fmeval.exceptions import EvalAlgorithmClientError
+from fmeval.transforms.transform import Transform
+from fmeval.util import require
+
+NestedTransform = Union[Transform, "TransformPipeline"]
+
+
+class TransformPipeline:
+    def __init__(self, nested_transforms: List[NestedTransform]):
+        require(
+            isinstance(nested_transforms, List),
+            "TransformPipeline initializer accepts a list containing Transforms or TransformPipelines.",
+        )
+        seen_keys = set()
+        duplicate_keys = []
```

Comment:
Suggestion: Should duplicate keys be a set instead? Having duplicates of a duplicate key in the list may not be helpful to the user and might instead be more of a nuisance for the user to dig through the different unique key names that are duplicated.

Reply:
Yeah, true. Actually, I think it would be better to maintain a list of `Transform` instances that contain duplicate keys, instead of the duplicate keys themselves.